src/redfishpower/redfishpower.c

```diff
@@ -754,6 +757,112 @@ static void setpowerpath(char **av, char **path, char **postdata)
     }
 }
 
+static void remove_initial_plugs(void)
+{
+    hostlist_iterator_t itr;
+    char *hostname;
+
+    if (!initial_plugs_setup)
+        return;
+
+    if (!(itr = hostlist_iterator_create(hosts)))
+        err_exit(true, "hostlist_iterator_create");
+
+    while ((hostname = hostlist_next(itr))) {
+        plugs_remove(plugs, hostname);
+        free(hostname);
+    }
+
+    hostlist_iterator_destroy(itr);
+    initial_plugs_setup = 0;
+    return;
+}
+
+static void setplugs(char **av)
+{
+    hostlist_t lplugs = NULL;
+    hostlist_t indices = NULL;
+    int plugcount, indexcount;
+    hostlist_iterator_t lplugsitr = NULL;
+    hostlist_iterator_t indicesitr = NULL;
+    char *plug;
+    char *indexstr;
+
+    if (!av[0] || !av[1]) {
+        printf("Usage: setplugs <plugnames> <hostindices>\n");
+        return;
+    }
+
+    if (!(lplugs = hostlist_create(av[0]))) {
+        printf("illegal plugs input\n");
```

Comment:
would it be handy to prefix all the setplugs failure messages with `setplugs:` to disambiguate these failures from others involving plugs?  E.g. `setplugs: malformed pluglist`